### ISSUE TITLE
Missing space reported in if [[ $var == \[*] ]]

### ISSUE THREAD

**Issue by twouters** [koalaman/shellcheck#945]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1020
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
var="[test]"
if [[ $var == \[*] ]]; then
    echo "square brackets around string"
fi
```

#### Here's what shellcheck currently says:

```
^-- SC1009: The mentioned parser error was in this if expression.
   ^-- SC1073: Couldn't parse this test expression.
                  ^-- SC1020: You need a space before the ]].
                  ^-- SC1072: Missing space before ]. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

No errors should be reported, because there is no missing space here.
Can be worked around by using e.g. `\[*']'` or `'['*']'`.

**Comment by aadrian** [comment]
This is also triggered when using something like https://github.com/niieani/bash-oo-framework#named-parameters-in-functions

**Comment by iconoclasthero** [comment]
Current shellcheck.net:
```
$ shellcheck myscript
 
Line 1:
[[ "$t" = *\] ]]
^-- [SC1073](https://www.shellcheck.net/wiki/SC1073) (error): Couldn't parse this test expression. Fix to allow more checks.
             ^-- [SC1020](https://www.shellcheck.net/wiki/SC1020) (error): You need a space before the ]].
             ^-- [SC1072](https://www.shellcheck.net/wiki/SC1072) (error): Missing space before ]. Fix any mentioned problems and try again.
```
